server/bleep/src/state.rs

```diff
@@ -96,6 +96,8 @@ impl<T> Clone for PersistedState<T> {
 
 impl StateSource {
     pub(crate) fn set_default_dir(&mut self, dir: &Path) {
+        std::fs::create_dir_all(&dir).unwrap();
```

Comment:
We should add an informative error message here, printing `dir` in the event that this errors.